src/Distribution/Server/Features/PackageList.hs

```diff
@@ -222,7 +234,7 @@ listFeature CoreFeature{..}
                 let pkgs = PackageIndex.lookupPackageName index pkgname
                 case pkgs of
```

Comment:
this can case on `Data.NonEmpty.nonEmpty` instead. Then the `Just nonEmptyList` case can pass that list on to `constructItem`